src/edu/psu/sweng500/team8/gui/BoardGUI.java

```diff
@@ -0,0 +1,261 @@
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+	}
+	
+	public void populatePencilMark(GameSession gameSession) {
+
+		this.gameSession = gameSession;
+
+		blocks[0][0].populatePencilMark(gameSession.getGameBoard().getBlock(0),
+				gameSession);
+		blocks[0][1].populatePencilMark(gameSession.getGameBoard().getBlock(1),
+				gameSession);
+		blocks[0][2].populatePencilMark(gameSession.getGameBoard().getBlock(2),
+				gameSession);
+
+		blocks[1][0].populatePencilMark(gameSession.getGameBoard().getBlock(3),
+				gameSession);
+		blocks[1][1].populatePencilMark(gameSession.getGameBoard().getBlock(4),
+				gameSession);
+		blocks[1][2].populatePencilMark(gameSession.getGameBoard().getBlock(5),
+				gameSession);
+
+		blocks[2][0].populatePencilMark(gameSession.getGameBoard().getBlock(6),
+				gameSession);
+		blocks[2][1].populatePencilMark(gameSession.getGameBoard().getBlock(7),
+				gameSession);
+		blocks[2][2].populatePencilMark(gameSession.getGameBoard().getBlock(8),
+				gameSession);
+	}
+
+	public void updateSelectedCellFromHint(CellCoordinates cellCoordinates, int number) {
+		
+		if (null != selectedCell) {
+			/* existing selection, clear it */
+			selectedCell.unselect();			
+		}
+		
+		selectedCell = blocks[cellCoordinates.getBlockIndex()/3][cellCoordinates.getBlockIndex()%3].getSelectedCell(cellCoordinates);
+		selectedCell.updateSelectedCellFromHint(number);
+	}
+	
+	private FocusAdapter buildFocusAdapter() {
+
+		return new FocusAdapter() {
+			public void focusGained(FocusEvent focusEvent) {
+				cellGainedFocus(focusEvent);
+			}
+
+			public void focusLost(FocusEvent focusEvent) {
+				cellLostFocus(focusEvent);
+			}
+		};
+	}
+	
+	private void cellGainedFocus(FocusEvent focusEvent) {
+		CellGUI newSelectedCell = (CellGUI) ((JTextField) focusEvent.getSource())
+				.getParent().getParent();
+		
+		if (null != this.selectedCell) {
+			/* existing selection, clear it */
+			this.selectedCell.unselect();			
+		}
+
+		this.selectedCell = newSelectedCell;
+		this.selectedCell.selectCell();
+	}
+
+	private void cellLostFocus(FocusEvent focusEvent) {
+		CellGUI unselectedCell = (CellGUI) ((JTextField) focusEvent.getSource())
+				.getParent().getParent();
+		unselectedCell.cellLostFocus(focusEvent);
+	}	
+	
+	private MouseAdapter buildMouseAdapter() {
+		return new MouseAdapter() {
+			@Override
+			public void mouseClicked(MouseEvent mouseEvent) {
+				mouseClickedTask(mouseEvent);
+			}
+		};
+	}
+	
+	private void mouseClickedTask(MouseEvent mouseEvent) {
+		if (null != this.selectedCell) {
+			/* existing selection, clear it */
+			this.selectedCell.unselect();			
+		}
+		
+		CellGUI currentPencilMarkDisplayCell = (CellGUI)((JPanel)(((JLabel)(mouseEvent.getSource())).getParent())).getParent();
+		this.selectedCell = currentPencilMarkDisplayCell;
+		
+		currentPencilMarkDisplayCell.mouseClicked();
+	}
+	
+	/**/
+	public void highlightIncorrectCells(Set<Cell> incorrectCells) {

```

Comment:
ctrl+i for indentation
ctrl+shift+f and for formatting
will do that throughout the code as well.


On Sun, Jul 12, 2015 at 10:17 PM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/gui/BoardGUI.java
> <https://github.com/garnryang/Team8/pull/13#discussion_r34429713>:
>
> > +	}
> > +	
> > +	private void mouseClickedTask(MouseEvent mouseEvent) {
> > +		if (null != this.selectedCell) {
> > +			/* existing selection, clear it */
> > +			this.selectedCell.unselect();			
> > +		}
> > +		
> > +		CellGUI currentPencilMarkDisplayCell = (CellGUI)((JPanel)(((JLabel)(mouseEvent.getSource())).getParent())).getParent();
> > +		this.selectedCell = currentPencilMarkDisplayCell;
> > +		
> > +		currentPencilMarkDisplayCell.mouseClicked();
> > +	}
> > +	
> > +	/**/
> > +	public void highlightIncorrectCells(Set<Cell> incorrectCells) {
>
> Fix tab formatting.
>
> Suggest just doing a select all then ctrl+i in Eclipse (autoformat)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/13/files#r34429713>.
>